contents/blog/the-posthog-array-1-41-0.md

```diff
@@ -0,0 +1,168 @@
+---
+date: 2022-11-02
+title: "Array 1.41.0: Improving performance by up to 400%"
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+categories: ['Release notes', 'Product updates']
+featuredImage: ../images/blog/posthog-array-blog.png
+featuredImageType: full
+---
+
+Want to know more about what we're up to? [Subscribe to HogMail, our newsletter](/newsletter), which we send every two weeks!
+
+## 1.41.0 update guide for self-hosted users
+Due to changes in this update, it is important check `alias` usage before [upgrading PostHog](/docs/runbook/upgrading-posthog) to 1.41.0 on a self-hosted instance. For example, assuming `email` is used as the identified user id, then: 
+
+```
+identify(email)                                   # in the frontend
+alias(backend_unique_id, email)      # in the backend - THIS WILL NOT WORK
+alias(email, backend_unique_id2)    # in the backend - the right order
+```
+> **If you haven't run async migrations 0005, 0006 and/or 0007 before:** Before updating to 1.41.0, check [ingestion warnings](#new-ingestion-warnings) and solve any outstanding issues. Run async migrations 0005, 0006 and/or 0007 at `<your-posthog-site>/instance/async_migrations`. 
```

Comment:
This is important, they shouldn't run the async migrations on the old version & we only care about 0007 being completed of not, but they might have 0005 and/or 0006 in either state.
```suggestion
> **If you haven't run async migration 0007 before:** Before updating to 1.41.0, check [ingestion warnings](#new-ingestion-warnings) and solve any outstanding issues. Then upgrade to 1.41.0 and then run async migration 0007 at `<your-posthog-site>/instance/async_migrations` (if you haven't ran 0005 yet start with that, then 0006 and finally 0007). 
```